crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs

```diff
@@ -2302,7 +2302,60 @@ impl Analyzer<'_, '_> {
         if let Type::Union(Union { types, .. }) = origin_ty.normalize() {
             for ty in types {
                 match ty.normalize() {
-                    Type::TypeLit(tl @ TypeLit { members, .. }) => {
+                    Type::Interface(interface) => {
```

Comment:
As this logic is duplicate of the logic below, I think it's better to extract this as a method and call it

Reply:
I think `get_additional_exclude_target_for_type_lit`  is the ideal name

Reply:
thx, i modifyed it